common/components/PrintableWallet/index.jsx

```diff
@@ -24,61 +25,58 @@ export default class PrintableWallet extends Component {
   };
 
   componentDidMount() {
-    // Start generating QR Codes as soon as component is ready
-    const opts = {
-      color: {
-        dark: '#000',
-        light: '#fff'
-      },
-      margin: 0,
-      errorCorrectionLevel: 'H'
-    };
-
-    QRCode.toDataURL(this.props.privateKey, opts, (err, url) => {
-      this.setState({ qrCodePkey: url });
-    });
+    // Start generating QR codes immediately
+    this._generateQrCode(this.props.privateKey, 'qrCodePkey');
+    this._generateQrCode(this.props.address, 'qrCodeAddress');
+  }
 
-    QRCode.toDataURL(this.props.address, opts, (err, url) => {
-      this.setState({ qrCodeAddress: url });
-    });
+  componentWillReceiveProps(nextProps) {
+    // Regenerate QR codes if props change
+    if (nextProps.privateKey !== this.props.privateKey) {
+      this._generateQrCode(nextProps.privateKey, 'qrCodePkey');
+    }
+    if (nextProps.address !== this.props.address) {
+      this._generateQrCode(nextProps.address, 'qrCodeAddress');
+    }
   }
 
-  print = () => {
-    const popup = window.open(
-      'about:blank',
-      'printWalletWindow',
-      `width=${walletWidth + 60},height=${walletHeight + 60},scrollbars=no`
+  _generateQrCode(value, stateKey) {
+    QRCode.toDataURL(
+      value,
+      {
+        color: {
+          dark: '#000',
+          light: '#fff'
+        },
+        margin: 0,
+        errorCorrectionLevel: 'H'
+      },
+      (err, url) => {
+        this.setState({ [stateKey]: url });
+      }
```

Comment:
Side note: I feel like `prettier` really butchered this. I had typed out:
```
    QRCode.toDataURL(value, {
      color: {
        dark: '#000',
        light: '#fff'
      },
      margin: 0,
      errorCorrectionLevel: 'H'
    }, (err, url) => {
      this.setState({ [stateKey]: url });
    });
```
Is this really preferable?